sql/sql_table.cc

```diff
@@ -3086,6 +3086,30 @@ bool mysql_rm_table_no_locks(THD *thd, TABLE_LIST *tables, bool if_exists,
 
   if (rm_table_sort_into_groups(thd, &drop_ctx, tables)) return true;
 
+  if (thd->variables.binlog_ddl_skip_rewrite) {
+    size_t table_count = 0;
+    for (TABLE_LIST *table = tables; table; table = table->next_local) {
+      ++table_count;
+      if (table_count > 1) break;
+    }
+
+    /*
+      When 'binlog_ddl_skip_rewrite' option is enabled
+      logging query as is for multi table drop statements
+      does not not work as expected if tables contain both normal tables
+      and temperary tables,consider these two cases.
+      Case1: Statement's like 'drop table t1,t2' where t1 is normal table
+      but t2 is temperory table  will fail on slave while applying log.
```

Comment:
temporary

Reply:
and t2 is a temporary table, will fail on the slave because temporary table will not be present on the slave.